spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -245,6 +251,22 @@ public static Expression convert(Predicate predicate) {
     return null;
   }
 
+  private static Pair<UnboundTerm<Object>, Object> predicateChildren(Predicate predicate) {
+    Object value;
+    UnboundTerm<Object> term;
```

Comment:
Optional, my thought is , it would be clearer to have term be of type 'NamedReference', instead of super class 'UnboundTerm', not sure what you think.

Reply:
I would like this to be `UnboundTerm ` because it could be a `UnboundTransform` as well in the following PRs.